### DISCUSSION TITLE
Improve map generator UI

### DISCUSSION THREAD

**Discussion by AaronVanGeffen** [OpenRCT2/OpenRCT2#22719] [Feature requests] [answered]
OpenRCT2 added a map generator to the scenario editor as early as 2014. Over the years, it has been improved with e.g. automatic tree placement, as well as support for loading height maps from PNGs. However, I feel the UI has been neglected in the process, mostly focusing on the technical feats. 

Consider screenshots of the current implementation:

![openrct2_map_gen_0](https://github.com/user-attachments/assets/e2731ee0-adb4-41d9-a997-b446d2faa860) ![openrct2_map_gen_1](https://github.com/user-attachments/assets/918789be-1f26-4f64-985e-a48888aa0912) ![openrct2_map_gen_2](https://github.com/user-attachments/assets/52456211-5721-4a82-987c-a73fa053883e) ![openrct2_map_gen_3](https://github.com/user-attachments/assets/56334b64-0b9d-443f-a92d-169bd86eca5d)

Note there are four tabs:
1. Basic map generator
2. Tree placement options (global)
3. Simplex map generator
4. PNG height map generator

One of these tabs affects the other three, while the others partially overlap. For example, the terrain choices exist in tabs 1 and 3, water level appears in tabs 1, 3, and 4, and finally the tree placement options appear in both tab 2 and 3.

I would like to propose we adopt a UI similar to the one in OpenLoco, which was initially based on the Locomotion landscape generator:

![openloco_landscape_gen_0a](https://github.com/user-attachments/assets/a19d924f-55a3-4594-bf69-0d594e7f7ad7) ![openloco_landscape_gen_0b](https://github.com/user-attachments/assets/59f038f4-370b-44c1-b807-1c3fd2e23f4f) ![openloco_landscape_gen_1](https://github.com/user-attachments/assets/a1bb9230-4bc5-4614-929a-11073e64d9ec) ![openloco_landscape_gen_2](https://github.com/user-attachments/assets/02192680-d4ed-4bb4-97f2-ce110ca2ce68) ![openloco_landscape_gen_3](https://github.com/user-attachments/assets/13961509-bacb-4f62-bb5e-54ad04a17206)

Note how the first tab allows selecting the generator type (original, simplex, PNG), and the other tabs each have a selection of options that are used by the selected generator. In doing so, there is plenty of space for more fine-tuning of the maps generated, which has led to more options having been implemented.

For OpenRCT2, I would like to propose the following steps:

- Introduce an 'initial' tab for the map size and generator selection.
  - This would include the Simplex and PNG settings, with only the relevant settings being visible. 
  - This could eventually include #15767 as well.
- Add a terrain tab for the current terrain options, and only show them there.
- Add a water tab for the current water options, and only show them there.
- Remove the tabs for the individual generators.
- Gradually port more features from the OpenLoco landscape generator after the initial PR.

**Comment by J-PIE-314** [comment, marked as answer]
Work on this is being done in PR #22735.
